src/XR/features/WebXREyeTracking.ts

```diff
@@ -0,0 +1,151 @@
+import { WebXRFeaturesManager, WebXRFeatureName } from '../webXRFeaturesManager';
+import { WebXRAbstractFeature } from './WebXRAbstractFeature';
+import { WebXRSessionManager } from '../webXRSessionManager';
+import { Observable } from "../../Misc/observable";
+import { Vector3, Quaternion } from "../../Maths/math.vector";
+import { Ray } from "../../Culling/ray";
+import { Nullable } from "../../types";
+
+/**
+ * The WebXR Eye Tracking feature grabs eye data from the device and provides it in an easy-access format.
+ * Currently only enabled for BabylonNative applications, developers intending to use this must enable the Gaze Input capability in their appx package manifest for data to flow.
+ */
+export class WebXREyeTracking extends WebXRAbstractFeature {
+    private _latestEyeSpace: Nullable<XRSpace>;
+    private _gazeRay: Nullable<Ray>;
+
+    private _tmpQuat: Quaternion = new Quaternion();

```

Comment:
Instead of creating a new temp quaternion, could you just use `TmpVectors.Quaternion[0]`?